pkg/controller/populators/forklift-populator.go

```diff
@@ -212,6 +207,10 @@ func (r *ForkliftPopulatorReconciler) reconcile(ctx context.Context, req reconci
 		res, err = r.reconcileCleanup(pvcPrime)
 	}
 
+	if pvcPrime.DeletionTimestamp != nil {
+		res, err = r.deletePopulatorPod(fmt.Sprintf("%s-%s", populatorPodPrefix, pvc.UID), pvc, pvcPrime)
```

Comment:
in what cases would pvc prime be deleted before the pod?

Reply:
`res, err = r.reconcileCleanup(pvcPrime)` in L#207 marks it for deletion

Reply:
Why not have `reconcileCleanup` delete the pod as well then?